lib/netext/httpext/error_codes.go

```diff
@@ -107,7 +108,25 @@ func http2ErrCodeOffset(code http2.ErrCode) errCode {
 
 // errorCodeForError returns the errorCode and a specific error message for given error.
 func errorCodeForError(err error) (errCode, string) {
-	switch e := errors.Cause(err).(type) {
+	inner := stderrors.Unwrap(err)
+	if inner != nil && inner != err {
+		code, resultErr := errorCodeForError(inner)
+		if code != defaultErrorCode {
+			return code, resultErr
+		}
+	}
```

Comment:
Is this code actually necessary? Judging by the the [code for `errors.Cause`](https://github.com/pkg/errors/blob/614d223910a179a466c1767a985424175c39b465/errors.go#L272-L288), I don't see a reason to do it like this

Reply:
What do you mean ... by is it necessary? If we have an error that has `Unwrap` method and one that has `Cause` ... well they don't work together, which is the whole point of this PR to "Unwrap" the corresponding errors ... correspondingly

Reply:
You still have the same `Unwrap()` code below, on the result of `errors.Cause(err)`, which [will be a no-op](https://github.com/pkg/errors/blob/614d223910a179a466c1767a985424175c39b465/errors.go#L272-L288) if the error wasn't wrapped with `github.com/pkg/errors`

Reply:
Yeah ... you are right :facepalm: , I first added this one and then ... figured out that I need the second one as well, but didn't think it through that ... well I wouldn't need the first one